aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -0,0 +1,1719 @@
+            inactive_shares,
+            pending_withdrawals: table::new<address, u64>(),
+            stake_pool_signer_cap,
+            total_coins_inactive: 0,
+            operator_commission_percentage,
+            add_stake_events: account::new_event_handle<AddStakeEvent>(&stake_pool_signer),
+            reactivate_stake_events: account::new_event_handle<ReactivateStakeEvent>(&stake_pool_signer),
+            unlock_stake_events: account::new_event_handle<UnlockStakeEvent>(&stake_pool_signer),
+            withdraw_stake_events: account::new_event_handle<WithdrawStakeEvent>(&stake_pool_signer),
+        });
+
+        // save delegation pool ownership and resource account address (inner stake pool address) on `owner`
+        move_to(owner, DelegationPoolOwnership { pool_address });
+    }
+
+    #[view]
+    public fun owner_cap_exists(addr: address): bool {
+        exists<DelegationPoolOwnership>(addr)
+    }
+
+    fun assert_owner_cap_exists(owner: address) {
+        assert!(owner_cap_exists(owner), error::not_found(EOWNER_CAP_NOT_FOUND));
+    }
+
+    #[view]
+    public fun get_owned_pool_address(owner: address): address acquires DelegationPoolOwnership {
+        assert_owner_cap_exists(owner);
+        borrow_global<DelegationPoolOwnership>(owner).pool_address
+    }
+
+    #[view]
+    public fun delegation_pool_exists(addr: address): bool {
+        exists<DelegationPool>(addr)
+    }
+
+    fun assert_delegation_pool_exists(pool_address: address) {
+        assert!(delegation_pool_exists(pool_address), error::invalid_argument(EDELEGATION_POOL_DOES_NOT_EXIST));
+    }
+
+    /// Retrieves the shared resource account owning the stake pool in order
+    /// to forward a stake-management operation to this underlying pool.
+    fun retrieve_stake_pool_signer(pool: &DelegationPool): signer {
+        account::create_signer_with_capability(&pool.stake_pool_signer_cap)
+    }
+
+    /// Get the address of delegation pool reference `pool`.
+    fun get_pool_address(pool: &DelegationPool): address {
+        account::get_signer_capability_address(&pool.stake_pool_signer_cap)
+    }
+
+    #[view]
+    /// Return the index of current observed lockup cycle on delegation pool `pool_address`.
+    /// This represents the key into `inactive_shares` of the `pending_inactive` shares pool.
+    public fun observed_lockup_cycle(pool_address: address): u64 acquires DelegationPool {
+        assert_delegation_pool_exists(pool_address);
+        borrow_global<DelegationPool>(pool_address).observed_lockup_cycle
+    }
+
+    /// Allows an owner to change the operator of the underlying stake pool.
+    public entry fun set_operator(owner: &signer, new_operator: address) acquires DelegationPoolOwnership, DelegationPool {
+        let pool_address = get_owned_pool_address(signer::address_of(owner));
+        // synchronize delegation and stake pools before any user operation
+        synchronize_delegation_pool(pool_address);
```

Comment:
Is this required primarily because of commission distribution? If so, can you clarify in the comment? My first impression is that this is not needed for just setting the operator, but then realized this is needed so the old operator gets their unpaid commission converted to shares. May be it's worth adding a test case for that as well?

Reply:
I agree, will explicitly describe the purpose of this sync call and add a test scenario